dora/src/bytecode/data.rs

```diff
@@ -277,6 +278,43 @@ pub enum BytecodeOpcode {
     RetPtr,
 }
 
+impl BytecodeOpcode {
+    pub fn need_position(&self) -> bool {
+        match *self {
+            BytecodeOpcode::InvokeDirectVoid
+            | BytecodeOpcode::InvokeDirectBool
+            | BytecodeOpcode::InvokeDirectByte
+            | BytecodeOpcode::InvokeDirectChar
+            | BytecodeOpcode::InvokeDirectInt
+            | BytecodeOpcode::InvokeDirectLong
+            | BytecodeOpcode::InvokeDirectFloat
+            | BytecodeOpcode::InvokeDirectDouble
+            | BytecodeOpcode::InvokeDirectPtr
+            | BytecodeOpcode::InvokeVirtualVoid
+            | BytecodeOpcode::InvokeVirtualBool
+            | BytecodeOpcode::InvokeVirtualByte
+            | BytecodeOpcode::InvokeVirtualChar
+            | BytecodeOpcode::InvokeVirtualInt
+            | BytecodeOpcode::InvokeVirtualLong
+            | BytecodeOpcode::InvokeVirtualFloat
+            | BytecodeOpcode::InvokeVirtualDouble
+            | BytecodeOpcode::InvokeVirtualPtr
+            | BytecodeOpcode::InvokeStaticVoid
+            | BytecodeOpcode::InvokeStaticBool
+            | BytecodeOpcode::InvokeStaticByte
+            | BytecodeOpcode::InvokeStaticChar
+            | BytecodeOpcode::InvokeStaticInt
+            | BytecodeOpcode::InvokeStaticLong
+            | BytecodeOpcode::InvokeStaticFloat
+            | BytecodeOpcode::InvokeStaticDouble
+            | BytecodeOpcode::InvokeStaticPtr
+            | BytecodeOpcode::NewObject
+            | BytecodeOpcode::Throw => true,
```

Comment:
I think the opcodes DivInt, DivMod, LoadFieldXXX and StoreFieldXXX also need the position.

Reply:
Will add them. DivLong and ModLong as well. What about the other Div* and Mod*?

Reply:
Float division returns Infinity and does not fail, so we don't need the position.